compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -111,10 +111,18 @@ impl MetaVarExpr {
         Ok(rslt)
     }
 
-    pub(crate) fn ident(&self) -> Option<Ident> {
-        match *self {
-            MetaVarExpr::Count(ident, _) | MetaVarExpr::Ignore(ident) => Some(ident),
-            MetaVarExpr::Concat { .. } | MetaVarExpr::Index(..) | MetaVarExpr::Len(..) => None,
+    pub(crate) fn metavars<A>(&self, mut aux: A, mut cb: impl FnMut(A, &Ident) -> A) -> A {
```

Comment:
Could we get this signature?
This would make the logic a bit easier to read in consumer code.

```suggestion
    pub(crate) fn for_each_metavar(&self, mut cb: impl FnMut(&Ident)) {
```

Reply:
That was my first attempt. Unfortunately, `LockstepIterSize::with` takes ownership.